File: src/Application Specific/Classes/Entity Collections/Checklist/Section/SectionValidator.vb
Namespace Entity

Namespace Sections

Public Class SectionValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oSection As Section)

        'make sure that contact object is valid
        If oSection.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oSection.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
                End If

                If oSection.Description.Trim.Length = 0 Then
                    Me.AddCriticalMessage("Description Missing")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Checklist/Section/SectionValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Sections
    {
        // make sure that contact object is valid
        public class SectionValidator : BaseValidator
        {
            public void Validate(Section oSection)
            {

                // make sure that contact object is valid
                if (oSection.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oSection.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oSection.Description.Trim().Length == 0)
                {
                    AddCriticalMessage("Description Missing");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}